src/Spice86.Core/Emulator/InterruptHandlers/Bios/SystemBiosInt15Handler.cs

```diff
@@ -96,10 +107,65 @@ public void ToggleA20GateOrGetStatus(bool calledFromVm) {
     }
 
     /// <summary>
-    /// Reports extended memory size in AX.
+    /// Reports  number of contiguous KB starting at absolute address 0x100000 <br/>
+    /// CF is cleared if successful, set otherwise. <br/> <br/>
+    /// Error is 0x80 (invalid command) for the IBM PC, Tandy, and PC Junior platforms. <br/>
+    /// Error is 0x86 (function not supported) for the IBM PC XT, and IBM PS/2.
     /// </summary>
-    public void GetExtendedMemorySize() {
-        State.AX = 0;
+    /// <remarks>
+    /// TSRs which wish to allocate extended memory to themselves often hook
+    /// this call, and return a reduced memory size.
+    /// They are then free to use the memory between the new and old sizes at will. <br/><br/>
+    /// The standard BIOS only returns memory between 1MB and 16MB; use AH=0xC7 for memory beyond 16MB.
+    /// </remarks>
+    public void GetExtendedMemorySize(bool calledFromVm) {
+        if(_extendedMemoryManager is not null) {
```

Comment:
Shouldn't this be
`if(_extendedMemoryManager == null) {`?

Additionally, what would be the correct response if XMS is not supported (either because its null or disabled via parameter)?